internal/graph/iterator_stream.go

```diff
@@ -0,0 +1,133 @@
+package graph
+
+import (
+	"context"
+	"slices"
+
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+
+	"github.com/openfga/openfga/pkg/storage"
+)
+
+type iteratorStream struct {
+	idx    int
```

Comment:
it's a code smell to me that `idx` is never referenced in this file, it's referenced outside of it 🤔  it feels like we should add a method to `iteratorStream` to get or set this `idx`.